code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_abilities.dm

```diff
@@ -62,3 +74,193 @@
+		playsound(X, 'sound/effects/alien_footstep_charge1.ogg', 50)
+
+	for(var/mob/living/carbon/human/M in X.loc)
+		if(M.lying && M.stat != DEAD)
+			X.visible_message(SPAN_DANGER("[X] runs [M] over!"),
+				SPAN_DANGER("You run [M] over!")
+			)
+
+			M.apply_damage(momentum * 10)
+			animation_flash_color(M)
+
+	X.recalculate_speed()
+
+/datum/action/xeno_action/onclick/charger_charge/proc/handle_dir_change(datum/source, old_dir, new_dir)
+	SIGNAL_HANDLER
+	if(new_dir != charge_dir)
+		charge_dir = new_dir
+		if(momentum)
+			stop_momentum()
+
+/datum/action/xeno_action/onclick/charger_charge/proc/handle_river(datum/source, covered)
+	SIGNAL_HANDLER
+	if(!covered)
+		stop_momentum()
+
+/datum/action/xeno_action/onclick/charger_charge/proc/update_speed(mob/living/carbon/Xenomorph/X)
+	SIGNAL_HANDLER
+	X.speed += momentum * speed_per_momentum
+
+/datum/action/xeno_action/onclick/charger_charge/proc/stop_momentum(datum/source)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/Xenomorph/X = owner
+	if(momentum == max_momentum)
+		X.visible_message(SPAN_DANGER("[X] skids to a halt!"))
+
+	REMOVE_TRAIT(X, TRAIT_CHARGING, TRAIT_SOURCE_XENO_ACTION_CHARGE)
+	steps_taken = 0
+	momentum = 0
+	X.recalculate_speed()
+	X.update_icons()
+
+/datum/action/xeno_action/onclick/charger_charge/proc/lose_momentum(amount)
+	if(amount >= momentum)
+		stop_momentum()
+	else
+		momentum -= amount
+		var/mob/living/carbon/Xenomorph/X = owner
+		X.recalculate_speed()
+
+/datum/action/xeno_action/onclick/charger_charge/proc/handle_collision(mob/living/carbon/Xenomorph/X, atom/A)
+	SIGNAL_HANDLER
+	if(!momentum)
+		stop_momentum()
+		return
+
+	var/result = A.handle_charge_collision(X, src)
+	switch(result)
+		if(XENO_CHARGE_TRY_MOVE)
+			if(step(X, charge_dir))
+				return COMPONENT_LIVING_COLLIDE_HANDLED
+
+/datum/action/xeno_action/onclick/charger_charge/proc/start_charging(datum/source)
+	SIGNAL_HANDLER
+	steps_taken = steps_to_charge
+
+
+/datum/action/xeno_action/activable/tumble
+	name = "Tumble"
+	action_icon_state = "tumble"
+	macro_path = /datum/action/xeno_action/verb/verb_crusher_tumble
+	action_type = XENO_ACTION_CLICK
+	ability_primacy = XENO_PRIMARY_ACTION_2
+
+	plasma_cost = 25
+	xeno_cooldown = 10 SECONDS
+
+/datum/action/xeno_action/activable/tumble/proc/handle_mob_collision(mob/living/carbon/human/H)
+	var/mob/living/carbon/Xenomorph/X = owner
+
+	X.visible_message(SPAN_XENODANGER("[X] Sweeps to the side, knocking down [H]!"), SPAN_XENODANGER("You knock over [H] as you sweep to the side!"))
+
+	var/turf/target_turf = get_turf(H)
+	xeno_throw_human(H, X, get_dir(X, H), 1)
+	H.apply_damage(15,BRUTE)
+	H.KnockDown(1)
+	playsound(H,'sound/weapons/alien_claw_block.ogg', 50, 1)
+	if(!LinkBlocked(X, get_turf(X), target_turf))
+		X.forceMove(target_turf)
+
+/datum/action/xeno_action/activable/tumble/proc/on_end_throw(start_charging)
+	var/mob/living/carbon/Xenomorph/X = owner
+	X.flags_atom &= ~DIRLOCK
+	if(start_charging)
+		SEND_SIGNAL(X, COMSIG_XENO_START_CHARGING)
+
```

Comment:
Same thing here and anywhere else you have 1 letter variable names that you've defined